### PR TITLE
4191 beehiiv actions

### PR DESCRIPTION
**Pull request by alysonturing**
### New actions:
- Create subscriber
- List all publications

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4276"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/beehiiv/app/beehiiv.app.ts [hunk 1]
```diff
@@ -1,13 +1,100 @@
...
+      };
+    },
+    _getRequestParams(opts: any) {
+      return {
+        ...opts,
+        url: this._getUrl(opts.path),
+        headers: this._getHeaders(),
+      };
+    },
+    filterEmptyValues(obj) {
```
Comment (andrewjschuang) [inline components/beehiiv/app/beehiiv.app.ts:58]: This method shouldn't be needed as `axios` already clears empty values. [See here](https://github.com/PipedreamHQ/pipedream/blob/66be3412a0aff553b60efbff5912237904eaf2c3/platform/lib/axios.ts#L53)

**Review by andrewjschuang — CHANGES_REQUESTED** [review]
LGTM aside for a minor comment below.

Also, I fixed the `package.json` dependencies, please see my commits for more information.

File: components/beehiiv/app/beehiiv.app.ts (hunk 1 above)
Comment (alysonturing) [reply components/beehiiv/app/beehiiv.app.ts:58]: it seems that it only removes "undefined" props in the object, but it seems to work on this scenario

**Review by andrewjschuang — APPROVED** [review]

**Comment by andrewjschuang** [comment]
LGTM!

**Comment by DilanAthukorala** [comment]
Verification completed, this is ready for release.

**Comment by ejwhite7** [comment]
Hey team - there seems to an issue - I'm getting these errors: 
Screenshot 2022-09-20 at 5 33 32 PM
Screenshot 2022-09-20 at 5 33 23 PM

**Comment by alysonturing** [comment]
> Hey team - there seems to an issue - I'm getting these errors:
> Screenshot 2022-09-20 at 5 33 32 PM
> Screenshot 2022-09-20 at 5 33 23 PM


It seems that you don’t have any publications… in order to list them, you must to have at least one registered on your account… please let me know if that works

**Comment by ejwhite7** [comment]
I do have a publication registered: 
Screenshot 2022-09-20 at 8 46 20 PM

**Comment by alysonturing** [comment]
@ejwhite7 ok, I'm testing it again, will be back with more information soon..

**Comment by alysonturing** [comment]
@ejwhite7 can you please try the action `List All Publications` and show me your results?
image

**Comment by ejwhite7** [comment]
That seems to have worked - thank you!

I’ll film a tutorial and get your integration posted on our site.

**Comment by alysonturing** [comment]
@ejwhite7 may you please check if your api key is correct? http error with status 401 normally represents unauthorized requests.
I was able to reproduce the error using an incomplete api key.

**Comment by alysonturing** [comment]
@ejwhite7 
ah, I just noticed that you said that it worked! thank you, and I'm sorry for any inconvenience...

**Comment by dannyroosevelt** [comment]
What was the issue?

**Comment by ejwhite7** [comment]
Evidently I needed to update my API key 🤦🏻‍♂️ Sent from my iPhoneOn Sep 20, 2022, at 9:11 PM, Danny Roosevelt ***@***.***> wrote:﻿
What was the issue?

**Comment by dannyroosevelt** [comment]
Ok, good to know -- we're working on reducing issues like this and hopefully making the errors more clear moving forward. 

Thanks for debugging @alysonfreitas!

@ejwhite7 let us know when you get the tutorial posted, so we can re-share to our community as well!

**Comment by ejwhite7** [comment]
The only thing that tripped me up was having to add the intermediate step to ‘list all publications’ drippy before setting up the ‘create new subscriber’ activity. Most other tools have the select all publications built-in so you can select the publication directly from the activity.Great work, and thank you for your help and for your patience.Best,EJSent from my iPhoneOn Sep 20, 2022, at 9:25 PM, Danny Roosevelt ***@***.***> wrote:﻿
Ok, good to know -- we're working on reducing issues like this and hopefully making the errors more clear moving forward.
Thanks for debugging @alysonfreitas!
@ejwhite7 let us know when you get the tutorial posted, so we can re-share to our community as well!

**Pull request merged**